connections/tests/utils.py

```diff
@@ -12,29 +13,94 @@ def make_apartments_sold_in_elastic() -> None:
     sleep(3)
 
 
-def get_elastic_apartments_for_sale_uuids() -> list:
+def get_elastic_apartments_for_sale_uuids(
+    etuovi=False, oikotie=False, only=False
+) -> list:
     s_obj = (
         Search()
         .query("match", _language="fi")
         .query("match", apartment_state_of_sale=ApartmentStateOfSale.FOR_SALE)
     )
+
+    if oikotie:
+        s_obj = s_obj.query("match", publish_on_oikotie=oikotie)
+        if only:
+            s_obj = s_obj.query("match", publish_on_etuovi=False)
+    if etuovi:
+        s_obj = s_obj.query("match", publish_on_etuovi=etuovi)
+        if only:
+            s_obj = s_obj.query("match", publish_on_oikotie=False)
+
+    if only and not etuovi and not oikotie:
+        s_obj = s_obj.query("match", publish_on_oikotie=oikotie).query(
+            "match", publish_on_etuovi=etuovi
+        )
+
+    if etuovi and oikotie and only:
+        # get apartments only by published flag
+        s_obj = (
+            Search()
+            .query("match", publish_on_oikotie=oikotie)
+            .query("match", publish_on_etuovi=etuovi)
+            .query("match", apartment_state_of_sale=ApartmentStateOfSale.RESERVED)
+        )
+
     s_obj.execute()
     scan = s_obj.scan()
     uuids = []
     for hit in scan:
-        uuids.append(str(hit.uuid))
+        uuids.append(hit.uuid)
     return uuids
 
 
 def get_elastic_apartments_for_sale_project_uuids() -> list:
+    """Used only in oikotie tests for fetching expected housing companies"""
     s_obj = (
         Search()
         .query("match", _language="fi")
         .query("match", apartment_state_of_sale=ApartmentStateOfSale.FOR_SALE)
+        .query("match", publish_on_oikotie=True)
     )
     s_obj.execute()
     scan = s_obj.scan()
     uuids = []
     for hit in scan:
         uuids.append(str(hit.project_uuid))
     return uuids
+
+
+def publish_elastic_apartments(uuids: list, etuovi=False, oikotie=False) -> list:
```

Comment:
I need more information about this function, let's check that on call too.

Reply:
This is done to see that changing publish_on_etuovi/oikotie flag is making correct updates to database